dvc/repo/add.py

```diff
@@ -34,12 +36,19 @@ def add(repo, target, recursive=False, no_commit=False, fname=None):
             )
         )
 
-    stages = _create_stages(repo, targets, fname, no_commit)
+    with repo.state:
+        stages = _create_stages(repo, targets, fname)
+
+        repo.check_dag(stages)
```

Comment:
probably, not related to this PR - would be great if add can do add modify and check graph dynamically, w/o recalculating it again on every add ... what about -R, btw - will it force recalculation on every single file in the directory?

Reply:
So for `-R` it won't recalculate it multiple times, as `_find_all_targets(repo, target, recursive) ` above will handle all of that at the same time. That being said, you have a good point about `check_dag` not really needing to re-collect graph in these cases, I'll double-check that.